src/service.py

```diff
@@ -49,24 +51,71 @@ def configure(self, snap_config: dict[str, Any]) -> None:
         self.snap_client.set(snap_config, typed=True)
 
 
-def snap_install(resource: str) -> SnapService:
-    """Install or refresh the snap, and return the snap service.
+def snap_install(resource: Optional[str], channel: str) -> SnapService:
```

Comment:
non-blocking: I know these functions were already here, but they do look odd. Given the mostly OO design of this code these utility functions feel more they would belong as methods of a Snap class.

Reply:
reworking these into a Snap class (which could be done in a separate PR) could also help resolve the awkwardness of having both a resource path and a channel in the same signature (channel and path would be class attributes and install would inspect them to determine how to behave)